js/src/run_trees.ts

```diff
@@ -733,6 +734,9 @@ export class RunTree implements BaseRun {
   }
 
   async postRun(excludeChildRuns = true): Promise<void> {
+    if (this._awaitInputsOnPost) {
```

Comment:
_question_: curious, do we actually need to have this flag? Why couldn't we either detect if it's a promise object or just always await?